oak-run-commons/src/main/java/org/apache/jackrabbit/oak/index/IndexImporterSupportBase.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.jackrabbit.oak.index;
+
+import org.apache.jackrabbit.oak.api.CommitFailedException;
+import org.apache.jackrabbit.oak.plugins.index.IndexEditorProvider;
+import org.apache.jackrabbit.oak.plugins.index.importer.AsyncIndexerLock;
+import org.apache.jackrabbit.oak.plugins.index.importer.ClusterNodeStoreLock;
+import org.apache.jackrabbit.oak.plugins.index.importer.IndexImporter;
+import org.apache.jackrabbit.oak.spi.state.Clusterable;
+import org.apache.jackrabbit.oak.spi.state.NodeStore;
+
+import java.io.File;
+import java.io.IOException;
+
+public abstract class IndexImporterSupportBase {
+
+    protected final NodeStore nodeStore;
+    protected final IndexHelper indexHelper;
+
+    public IndexImporterSupportBase(IndexHelper indexHelper) {
+        this.nodeStore = indexHelper.getNodeStore();
+        this.indexHelper = indexHelper;
+    }
+
+    public void importIndex(File importDir) throws IOException, CommitFailedException {
+        IndexImporter importer = new IndexImporter(nodeStore, importDir, createIndexEditorProvider(), createLock());
+        addImportProviders(importer);
+        importer.importIndex();
+    }
+
+    private AsyncIndexerLock createLock() {
+        if (nodeStore instanceof Clusterable) {
```

Comment:
I generally try to avoid using "instanceof"... I see the same "instanceof" check is used in oak-core IndexerMBeanImpl.createLock. I think it is OK. The alternative would be to either change the NodeStore interface, but that's not nice either; or add a method to ClusterNodeStoreLock to do the instanceof check there... but both isn't much better.

Reply:
Actually this code was as it is - I just moved it out to a base class so that it can be used in oak-run and oak-run-elastic both.